### PR TITLE
rnpm-plugin-windows fails when using npm

### LINKED ISSUE #3862: react-native windows doesn't work after installing windows plug-in using npm
Our Getting Started guide says that we still support installing rnpm-plugin-windows via npm but it is currently broken.

REPRO STEPS:
react-native init myproject --version 0.60.0
cd myproject
npm install --save rnpm-plugin-windows
react-native windows --template vnext

RESULT
warn Package rnpm-plugin-windows has been ignored because it contains invalid configuration. Reason: Cannot find module '@react-native-community/cli/build/tools/generator/promptSync'
error Unrecognized command "windows".

### PR DESCRIPTION
**Pull request by acoates-ms**
Fixes #3862

rnpm-plugin-windows was using a dependency that it wasn't declaring.  Rather than having rnpm declare a specific react-native-community\cli dependency, which can cause issues since different RN versions require different versions of the CLI.  I changed the dependency to use another package.

This also changes the CLI to only prompt the vnext/legacy prompt when used against RN 0.59.  Since thats the only version that has a good legacy and vnext version.

If the user it using >=0.60 we will automatically use the vnext version.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3998)

### CODE REVIEW

File: current/local-cli/rnpm/windows/src/windows.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ const Common = require('./common');
 const chalk = require('chalk');
 const execSync = require('child_process').execSync;
 const path = require('path');
-const prompt = require('@react-native-community/cli/build/tools/generator/promptSync').default();
```
Comment (kmelmon) [inline current/local-cli/rnpm/windows/src/windows.js]: local-cli\generator-common\index.js also require's and uses this module.  Does that need to be changed as well?  prompts looks better anyway.

Comment (acoates-ms) [reply current/local-cli/rnpm/windows/src/windows.js]: Thats in react-native-windows which declares the cli dep already.

Comment (kmelmon) [reply current/local-cli/rnpm/windows/src/windows.js]: That dependency is about to be removed when we move to 61

Comment (acoates-ms) [reply current/local-cli/rnpm/windows/src/windows.js]: If its removed the generator will have to be updated to not use it.

I expect that over time we will require the cli dependency as we work to properly implement auto-linking etc -- so maybe we should just keep the dependency and update it to match the same version range that react-native itself uses in 61.

**Pull request merged**